### PR TITLE
Update Stacks2

### PR DESCRIPTION
**Pull request by gallardoalba**
FOR CONTRIBUTOR:
* [x] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [x] - License permits unrestricted use (educational + commercial)
* [ ] - This PR adds a new tool or tool collection
* [x] - This PR updates an existing tool or tool collection
* [ ] - This PR does something else (explain below)

This PR includes a functionality requested by a user in [GHelp](https://help.galaxyproject.org/t/stacks2-process-radtags-doesnt-have-a-no-barcode-option/7206), i.e. it includes the possibility of not using barcodes in *Stacks2 process radtags*.

In addition, this PR includes some additional modifications in the *Stacks2 process short-reads* wrapper in order to maintain the compatibility.

### CODE REVIEW

File: tools/stacks2/macros.xml [hunk 1]
```diff
@@ -217,6 +217,24 @@ Stacks was developed by Julian Catchen with contributions from Angel Amores, Pau
         </expand>
     </xml>
 
+
+    <xml name="barcode_selector">
```
Comment (bernt-matthias) [inline tools/stacks2/macros.xml:221]: I would prefer a solution without an additional conditional .. if possible.

Maybe we simply add an option `disable` to the `barcode_encoding` select and make the `barcode` parameter optional .. adding to the help that the barcode file is only used if `barcode_input != disabled`.

File: tools/stacks2/macros.xml [hunk 2]
```diff
@@ -2,15 +2,15 @@
 <macros>
     <xml name="requirements">
         <requirements>
-            <requirement type="package" version="@STACKS_VERSION@">stacks</requirement>
+            <requirement type="package" version="@TOOL_VERSION@">stacks</requirement>
```
Comment (bernt-matthias) [inline tools/stacks2/macros.xml:5]: That's a good idea. Then we should have 2.60 soon :)

**Review by bernt-matthias — CHANGES_REQUESTED** [review]
According to the source code this should also work for `shortreads`, so I would prefer to keep the macros.

File: tools/stacks2/macros_process.xml [hunk 3]
```diff
@@ -61,6 +38,7 @@
             <filter>capture is True and input_type['input_type_select'] == "paired"</filter>
             <expand macro="discover_faq_output_macro" pattern="(?P&lt;identifier_0&gt;.+)\.(?P&lt;identifier_1&gt;[^.]+)" dir="stacks_outputs/discarded"/>
         </collection>
+        <yield/>
```
[the remark this answers is not in the document]
Comment (gallardoalba) [reply tools/stacks2/macros_process.xml:41]: Done!

File: tools/stacks2/macros.xml (hunk 1 above)
Comment (gallardoalba) [reply tools/stacks2/macros.xml:221]: Done!

**Comment by bernt-matthias** [comment]
Thanks for working on this @gallardoalba .. once the tests are successful I would like to add the commits from https://github.com/galaxyproject/tools-iuc/pull/4194 .. is this fine for you?

**Comment by gallardoalba** [comment]
> Thanks for working on this @gallardoalba .. once the tests are successful I would like to add the commits from #4194 .. is this fine for you?

Sure. Thanks you so much for your review.

File: tools/stacks2/test-data/shortreads/PopA_01.fq [hunk 4]
```diff
@@ -1,640 +0,0 @@
-@lane1_fakedata0_0 1:N:0:/1
```
Comment (bernt-matthias) [inline tools/stacks2/test-data/shortreads/PopA_01.fq:1]: This test file should not become empty, or? 

Can we extend `test-data/gentest,sh` by calls to `shortreads` and `procrad` which generates these files and then reproduce it witch the Galaxy tools?

Comment (gallardoalba) [reply tools/stacks2/test-data/shortreads/PopA_01.fq:1]: Sure, I'll have a look at this.

**Comment by bernt-matthias** [comment]
I added my commits. Lets see if it works.

**Review by bgruening — APPROVED** [review]

**Review by bernt-matthias — APPROVED** [review]

**Comment by bernt-matthias** [comment]
@gallardoalba could you double check my changes before merging?

**Comment by gallardoalba** [comment]
> @gallardoalba could you double check my changes before merging?

It seems fine to me.

**Review by bernt-matthias — APPROVED** [review]

**Comment by bernt-matthias** [comment]
Cool. Thanks @gallardoalba

**Pull request merged**